setup.py

```diff
@@ -40,7 +37,42 @@
         "XlsxWriter>=3.0.3",
         # for tox
         "pluggy>=0.12.0",
-    ],
+    ]
+
+
+def _urllib3():
+    """Possibly constrain version of urllib3
+
+    Adapted from urllib3's checks for ssl
+    https://github.com/urllib3/urllib3/blob/2ac40569acb464074bdc3f308124d781d6aa0860/src/urllib3/__init__.py#L28
+    """
+
+    def _ssl_ok(ssl):
+        return (
+            # python ssl may not be compiled with OpenSSL (ex LibreSSL on MacOS).
+            # https://github.com/urllib3/urllib3/issues/3020
+            getattr(ssl, "OPENSSL_VERSION", "").startswith("OpenSSL ")
```

Comment:
Why make this a getattr? Wouldn't it be an ImportError for it to be undefined https://github.com/python/cpython/blob/v3.9.17/Lib/ssl.py#L101 ?

Reply:
`ssl` is defined. `OPENSSL_VERSION` may not be, depending on which version of `ssl` is being installed. If it  doesn't have an `OPENSSL_VERSION`, the new version of `urllib3` will fail to run (with an undefined attr) so we have to fall back.